modules/git/git.go

```diff
@@ -39,36 +39,37 @@ var (
 	gitVersion *version.Version
 )
 
-// loadGitVersion returns current Git version from shell. Internal usage only.
-func loadGitVersion() (*version.Version, error) {
+// loadGitVersion tries to get the current git version and stores it into a global variable
+func loadGitVersion() error {
 	// doesn't need RWMutex because it's executed by Init()
 	if gitVersion != nil {
-		return gitVersion, nil
+		return nil
 	}
 
 	stdout, _, runErr := NewCommand(DefaultContext, "version").RunStdString(nil)
 	if runErr != nil {
-		return nil, runErr
+		return runErr
 	}
 
-	fields := strings.Fields(stdout)
-	if len(fields) < 3 {
-		return nil, fmt.Errorf("invalid git version output: %s", stdout)
+	ver, err := parseGitVersionLine(strings.TrimSpace(stdout))
+	if err == nil {
+		gitVersion = ver
 	}
+	return err
+}
 
-	var versionString string
-
-	// Handle special case on Windows.
-	i := strings.Index(fields[2], "windows")
-	if i >= 1 {
-		versionString = fields[2][:i-1]
-	} else {
-		versionString = fields[2]
+func parseGitVersionLine(s string) (*version.Version, error) {
+	fields := strings.Fields(s)
+	if len(fields) < 3 {
+		return nil, fmt.Errorf("invalid git version: %q", s)
 	}
 
-	var err error
-	gitVersion, err = version.NewVersion(versionString)
-	return gitVersion, err
+	// version string is like: "git version 2.29.3" or "git version 2.29.3.windows.1"
+	versionString := fields[2]
+	if pos := strings.Index(versionString, "windows"); pos >= 1 {
```

Comment:
`>= 1`? Not `>= 0`?

Reply:
>= 0, because there is `versionString[:pos-1]` below.

Reply:
But we could use `>= 0` and `:pos`.

Reply:
That's from old code. I guess it might want to work with `2.3.4.windows` and `2.3.4-windows`

Reply:
Right, the -1 is needed for the prefix and therefore `i` must be `>=1`.